src/main/java/org/sagebionetworks/template/repo/agent/BedrockAgentContextProvider.java

```diff
@@ -12,23 +13,33 @@
 import org.sagebionetworks.template.config.RepoConfiguration;
 import org.sagebionetworks.template.repo.VelocityContextProvider;
 
+import com.amazonaws.services.s3.AmazonS3Client;
 import com.google.inject.Inject;
 
 public class BedrockAgentContextProvider implements VelocityContextProvider {
 	
 	private final RepoConfiguration repoConfig;
+	private final AmazonS3Client s3Cient;
 
 	@Inject
-	public BedrockAgentContextProvider(RepoConfiguration repoConfig) {
+	public BedrockAgentContextProvider(RepoConfiguration repoConfig, AmazonS3Client s3Client) {
 		super();
 		this.repoConfig = repoConfig;
+		this.s3Cient = s3Client;
 	}
 
 	@Override
 	public void addToContext(VelocityContext context) {
 		String stack = repoConfig.getProperty(PROPERTY_KEY_STACK);
 		String instance = repoConfig.getProperty(PROPERTY_KEY_INSTANCE);
 		String agentName = new StringJoiner("-").add(stack).add(instance).add("agent").toString();
+		
+		String openApiSchemaBucket = String.format("%s-configuration.sagebase.org", stack);
+		String openApiSchemakey = String.format("chat/openapi/%s/%s.json", instance, UUID.randomUUID().toString());
```

Comment:
Instead of the UUID, the stack + instance should be enough for defining a version.